webapp_v2/src/components/singlestudy/HomeView/StudyTreeView/index.tsx

```diff
@@ -0,0 +1,122 @@
+      <SVGRect key={`rect-for-name-deco-${i}-${j}`} x={RECT_WIDTH + RECT_X_SPACING} y={cy - TILE_SIZE_Y_2 + RECT_Y_SPACING_2} width={RECT_DECORATION} height={TILE_SIZE_Y - RECT_Y_SPACING} fill={hoverColor} />,
+      <SVGText key={`name-${i}-${j}`} x={RECT_WIDTH + RECT_X_SPACING + RECT_DECORATION + TEXT_SPACING} y={cy + RECT_Y_SPACING_2} fill={hoverId === id ? 'black' : 'white'} fontSize={TEXT_SIZE} onClick={() => onClick(id)} onMouseOver={(e) => onMouseOver(id)}>{name}</SVGText>];
+    if (verticalLineEnd > 0) { res.push(<path key={`verticalLine-${i}-${j}`} d={`M ${cx} ${cy + CIRCLE_RADIUS} L ${cx} ${verticalLineEnd}`} fill={verticalLineColor} stroke={verticalLineColor} strokeWidth={`${STROKE_WIDTH}`} />); }
+    if (i > 0) { res.push(<path key={`horizontalLine-${i}-${j}`} d={`M ${cx - CIRCLE_RADIUS - CURVE_OFFSET},${cy} C ${cx - TILE_SIZE_X},${cy} ${cx - TILE_SIZE_X},${cy} ${cx - TILE_SIZE_X},${cy - TILE_SIZE_Y + 2 * CIRCLE_RADIUS} M ${cx - CIRCLE_RADIUS},${cy} L ${cx - CIRCLE_RADIUS - CURVE_OFFSET},${cy}`} fill="transparent" stroke={color} strokeWidth={`${STROKE_WIDTH}`} />); }
+
+    let recursiveHeight = 1;
+    res = res.concat(children.map((elm, index) => {
+      if (index === 0) recursiveHeight = j + 1;
+      else recursiveHeight = recursiveHeight + children[index - 1].drawOptions.nbAllChildrens + 1;
+      return buildRecursiveTree(elm, i + 1, recursiveHeight);
+    }));
+    return res;
+  };
+
+  const renderTree = (tree: StudyTree) : React.ReactNode => {
+    const { drawOptions } = tree;
+    const { depth, nbAllChildrens } = drawOptions;
+    return (
+      <svg viewBox={`0 0 ${Math.max(TILE_SIZE_X * (depth + DEPTH_OFFSET), MIN_WIDTH) + RECT_TEXT_WIDTH + RECT_X_SPACING} ${TILE_SIZE_Y * (nbAllChildrens + 1) + TILE_SIZE_Y_2}`}>
+        {buildRecursiveTree(tree, 0, 0) }
+      </svg>
+    );
+  };
+
+  useEffect(() => {
+    const buildStudyTree = async () => {
+      if (study && childrenTree) {
+        const tmp = await getTreeNodes(study, parents, childrenTree);
+        setStudyTree(tmp);
+      }
+    };
+    buildStudyTree();
+  }, [study, parents, childrenTree]);
+
+  return (
+    <Box display="flex" flexDirection="row" justifyContent="flex-start" alignItems="flex-start" sx={{ width: '100%', flexGrow: 1, overflowY: 'auto', ...scrollbarStyle }}>
+      <Box minWidth={treeWidth / ZOOM_OUT} minHeight={treeHeight / ZOOM_OUT} display="flex" flexDirection="column" justifyContent="flex-start" alignItems="flex-start">
+        {studyTree && renderTree(studyTree)}
+      </Box>
+      <Box height="100%" display="flex" flexDirection="column" justifyContent="flex-start" alignItems="center">
+        {studyTree && renderTree(studyTree)}
+      </Box>
```

Comment:
at first when I saw it I found it nice (like in vscode / sublime text) but this is useless. Also when you move the splitter... it growwwws